src/js/App/GlobalFilter/GlobalFilter.js

```diff
@@ -23,19 +136,22 @@ const GlobalFilter = () => {
   const [token, setToken] = useState();
   const dispatch = useDispatch();
   const { isLoaded, count, total, sapCount, isDisabled } = useSelector(
-    ({ globalFilter: { tags, sid, workloads, globalFilterHidden }, chrome: { appId } }) => ({
-      isLoaded: tags?.isLoaded && sid?.isLoaded && workloads?.isLoaded,
-      count: tags?.count || 0 + sid?.count || 0 + workloads?.count || 0,
-      total: tags?.total || 0 + sid?.total || 0 + workloads?.total || 0,
-      sapCount: workloads?.hasSap,
-      isDisabled: globalFilterHidden || !appId,
-    }),
+    ({ globalFilter: { tags, sid, workloads, globalFilterHidden }, chrome: { appId } }) => {
+      return {
+        isLoaded: tags.isLoaded && sid.isLoaded && workloads.isLoaded,
+        count: tags.count + sid.count + workloads.count,
+        total: tags.total + sid.total + workloads.total,
+        sapCount: workloads.hasSap,
+        isDisabled: globalFilterHidden || !appId,
+      };
+    },
     shallowEqual
   );
-  const tags = useSelector(({ globalFilter: { tags } }) => tags?.items || []);
-  const sid = useSelector(({ globalFilter: { sid } }) => sid?.items || []);
+  const tags = useSelector(({ globalFilter: { tags } }) => tags.items || [], shallowEqual);
+  const sid = useSelector(({ globalFilter: { sid } }) => sid.items || [], shallowEqual);
```

Comment:
Because items contain arrays I am not sure if shallowEqual will do the trick.

Reply:
It will, and here is why. The only time when the filter should not update is when it's initializing. meaning when comparing `[] === []`. Any change to global filter results in tags and sid re-fetch and we will have to update no matter what. Plus, unfortunately, these arrays are circular structures (containing react components and functions) so not event `lodash.isEqaul` gives you the correct information.

I want to address that in post-summit refactoring and get rid of any components lurking in state management